plugins/rbac/src/hooks/useRoles.ts

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { useAsync, useAsyncRetry, useInterval } from 'react-use';
+
+import { useApi } from '@backstage/core-plugin-api';
+
+import { Role, RoleBasedPolicy } from '@janus-idp/backstage-plugin-rbac-common';
+
+import { rbacApiRef } from '../api/RBACBackendClient';
+import { RolesData } from '../types';
+import { getPermissions } from '../utils/rbac-utils';
+
+export const useRoles = () => {
+  const rbacApi = useApi(rbacApiRef);
+  const {
+    loading: rolesLoading,
+    value: roles,
+    retry,
+  } = useAsyncRetry(async () => {
+    return await rbacApi.getRoles();
+  });
+
+  const { loading: policiesLoading, value: policies } = useAsync(
+    async () => await rbacApi.getPolicies(),
+    [],
+  );
+
+  const data: RolesData[] = React.useMemo(
+    () =>
+      !roles || roles.length === 0
+        ? []
+        : roles.reduce(
+            (acc: any, role: Role) => [
+              ...acc,
+              {
+                id: role.name,
+                name: role.name,
+                description: '-',
```

Comment:
shall we hardcode `-` or use `role.description` and once the backend produces the data it'll be handled? 

```suggestion
                description: role.description || '-',
```

same applies for `modifiedBy` and `lastModified`. If for some field backend will not provide data in near term then we should not show that column as well

cc @PatAKnight

Reply:
Ideally, we aim to release the update with `modifiedBy`, `description`, and `lastModified` as soon as we receive the green light to begin working on them. However, I believe it might be a good idea to temporarily omit the columns instead of leaving it blank or with the hardcoded `-`. This way, the feature remains functional before the next release, and we can potentially avoid complaints about missing data / empty data.

Reply:
Thanks @PatAKnight 

@parvathyvr FYI

Reply:
@debsmita1 what's the decision on this?

Reply:
@invincibleJai  omitted the field